### PR TITLE
fix: Refactored NavigationPills Functionality for fixing SRS-654

### PR DESCRIPTION
**Pull request by midhun-aot**
- Refactored NavigationPills Component
- Fixed issue reported with SRS-654

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Backend](https://nr-site-registry-208-backend.apps.silver.devops.gov.bc.ca)
- [Frontend](https://nr-site-registry-208-frontend.apps.silver.devops.gov.bc.ca)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-site-registry/actions/workflows/merge.yml)

### CODE REVIEW

**Review by bolny — APPROVED** [review]

File: frontend/src/app/components/navigation/navigationpills/NavigationPills.tsx [hunk 1]
```diff
@@ -1,50 +1,83 @@
...
 import useMediaQuery from '../../../hooks/useMediaQuery';
 
 const NavigationPills: React.FC<INavigationPills> = ({
-  items,
   components,
-  dropdownItems,
   isDisable,
 }) => {
-  const [activeIndex, setActiveIndex] = useState<number>(0);
+  const [activeTabKey, setactiveTabKey] = useState<string>('');
```
Comment (anton-bcgov) [inline frontend/src/app/components/navigation/navigationpills/NavigationPills.tsx:13]: would it make sense to have the default state value as `useState<string>(components[0].value)`? This way you can get rid of the useEffect on lines 43-50

**Comment by jaspalsingh-aot** [comment]
Please make sure to add the changes of INavigationPills and NavigationPills to NPM packages as well.

**Comment by midhun-aot** [comment]
> Please make sure to add the changes of INavigationPills and NavigationPills to NPM packages as well.

As already discussed yesterday I will add it after NPM packages are approved :)

File: frontend/src/app/components/navigation/navigationpills/NavigationPills.tsx (hunk 1 above)
Comment (midhun-aot) [reply frontend/src/app/components/navigation/navigationpills/NavigationPills.tsx:13]: I tried to set it as default value and works in most of the cases , but in 1 scenario it fails and do not render component. So keeping it same.

**Pull request merged**